### PR TITLE
DMP-54 Add Notification table

### PR DESCRIPTION
**Pull request by cmagowan-hmcts**
https://tools.hmcts.net/jira/browse/DMP-54

Create database table for notifications

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/notification/entity/Notification.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    private String status;
+
+    @Column(name = "attempts")
+    private int attempts;
+
+    @Column(name = "template_values")
+    private String templateValues;
+
+    @Column(name = "created_datetime")
+    private Timestamp createdDatetime;
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/darts/notification/entity/Notification.java:46]: Super minor but shouldn't 'Time' start with capital?

File: src/main/java/uk/gov/hmcts/darts/notification/service/impl/NotificationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import uk.gov.hmcts.darts.notification.service.NotificationService;
+
+import java.sql.Timestamp;
+
+
+@RequiredArgsConstructor
+@Service
+public class NotificationServiceImpl implements NotificationService {
+
+    @Autowired
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/notification/service/impl/NotificationServiceImpl.java:19]: This `@Autowired` can be dropped here if we are using constructor arg instantiation using the lombok annotation

File: src/main/resources/application-dev.yaml [hunk 3]
```diff
@@ -33,3 +33,18 @@
...
             token-uri: https://hmctsdartsb2csbox.b2clogin.com/hmctsdartsb2csbox.onmicrosoft.com/B2C_1_darts_externaluser_signin/oauth2/v2.0/token
             jwk-set-uri: https://hmctsdartsb2csbox.b2clogin.com/hmctsdartsb2csbox.onmicrosoft.com/B2C_1_darts_externaluser_signin/discovery/v2.0/keys
+
+ logging:
+   level:
+     uk.gov.hmcts.darts: DEBUG
+     org.springframework.web.servlet.mvc.method.annotation: DEBUG
+     org.springframework.web: DEBUG
+
+ spring.datasource.url: "jdbc:postgresql://localhost:5432/dartsdb"
```
Comment (richardbull) [inline src/main/resources/application-dev.yaml:43]: These properties probably need to be yamlified - or do we drop them for time being if not being used yet?
Dependent on DMP-45 build / infra stuff

File: src/main/java/uk/gov/hmcts/darts/notification/service/impl/NotificationServiceImpl.java (hunk 2 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/notification/service/impl/NotificationServiceImpl.java:19]: Not quite true as notificationRepo isn't marked as final, but i can do.

File: src/main/java/uk/gov/hmcts/darts/notification/entity/Notification.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
+package uk.gov.hmcts.darts.notification.entity;
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/notification/entity/Notification.java:1]: Wondering should we rename this to be `entities` to match the package under common or vice-versa.
build.gradle stuff too

File: src/main/java/uk/gov/hmcts/darts/notification/service/impl/NotificationServiceImpl.java (hunk 2 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/notification/service/impl/NotificationServiceImpl.java]: have added final and removed @Autowired

File: src/main/resources/application-dev.yaml (hunk 3 above)
Comment (cmagowan-hmcts) [reply src/main/resources/application-dev.yaml]: Hadn't realised some of these were set elsewhere, so i have removed them all for now, as Scott's work should supersede it anyway. 
not sure about yamlifing the spring.logging elements, as i find that they are much easier to reach when in the dot format. Happy to yamlify them for consistency. 
@jackmaloney any preference?

File: src/main/java/uk/gov/hmcts/darts/notification/entity/Notification.java (hunk 4 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/notification/entity/Notification.java:1]: I have it in my head that package should always be singular, but I think it is again down to personal perference. Either way, we should be consistent.
@jackmaloney another question for you if you have a preference?

File: src/main/java/uk/gov/hmcts/darts/notification/entity/Notification.java (hunk 1 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/notification/entity/Notification.java]: renamed column to created_date_time

File: src/main/resources/application-dev.yaml (hunk 3 above)
Comment (richardbull) [reply src/main/resources/application-dev.yaml]: I prefer the properties in dot notation - but honestly didn't think we could have this style in yaml file. Interesting mix. :)

File: src/main/java/uk/gov/hmcts/darts/notification/entity/Notification.java (hunk 4 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/notification/entity/Notification.java:1]: in fact, been looking at another project in hmcts, and they seem to be mostly singular also, so guess go with that.
image

i'll change entities to entity in common

File: src/main/resources/application-dev.yaml (hunk 3 above)
Comment (cmagowan-hmcts) [reply src/main/resources/application-dev.yaml]: yeah, works fine. ok, if you are happy i will resolve comment

**Review by richardbull — APPROVED** [review]

**Pull request merged**